ClientCachingProviderTest failures

Failing [tests in recent builds](
https://hazelcast-l337.ci.cloudbees.com/job/Hazelcast-3.x-freestyle/1495/#showFailuresLink):
```
com.hazelcast.client.cache.ClientCachingProviderTest.whenDefaultCacheManager_thenSharedNameHazelcastInstanceExists
com.hazelcast.client.cache.ClientCachingProviderTest.whenDefaultCacheManager_withUnnamedDefaultInstance_thenNoSharedNameHazelcastInstanceExists
```

with:

```
javax.cache.CacheException: Error opening URI [hazelcast]
	at com.hazelcast.cache.impl.AbstractHazelcastCachingProvider.getCacheManager(AbstractHazelcastCachingProvider.java:130)
	at com.hazelcast.cache.impl.AbstractHazelcastCachingProvider.getCacheManager(AbstractHazelcastCachingProvider.java:159)
	at com.hazelcast.cache.HazelcastCachingProvider.getCacheManager(HazelcastCachingProvider.java:174)
	at com.hazelcast.cache.CachingProviderTest.whenDefaultCacheManager_thenSharedNameHazelcastInstanceExists(CachingProviderTest.java:228)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:39)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:25)
	at java.lang.reflect.Method.invoke(Method.java:597)
	at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:50)
	at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12)
	at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:47)
	at org.junit.internal.runners.statements.InvokeMethod.evaluate(InvokeMethod.java:17)
	at com.hazelcast.test.FailOnTimeoutStatement$CallableStatement.call(FailOnTimeoutStatement.java:105)
	at com.hazelcast.test.FailOnTimeoutStatement$CallableStatement.call(FailOnTimeoutStatement.java:97)
	at java.util.concurrent.FutureTask$Sync.innerRun(FutureTask.java:303)
	at java.util.concurrent.FutureTask.run(FutureTask.java:138)
	at java.lang.Thread.run(Thread.java:662)
Caused by: java.lang.IllegalStateException: Unable to connect to any address! The following addresses were tried: [[127.0.0.1]:5703, [127.0.0.1]:5702, [127.0.0.1]:5701]
	at com.hazelcast.client.connection.nio.ClientConnectionManagerImpl.connectToClusterInternal(ClientConnectionManagerImpl.java:875)
	at com.hazelcast.client.connection.nio.ClientConnectionManagerImpl.access$1500(ClientConnectionManagerImpl.java:102)
	at com.hazelcast.client.connection.nio.ClientConnectionManagerImpl$3.call(ClientConnectionManagerImpl.java:885)
	at com.hazelcast.client.connection.nio.ClientConnectionManagerImpl$3.call(ClientConnectionManagerImpl.java:881)
	at java.util.concurrent.FutureTask$Sync.innerRun(FutureTask.java:303)
	at java.util.concurrent.FutureTask.run(FutureTask.java:138)
	at java.util.concurrent.ThreadPoolExecutor$Worker.runTask(ThreadPoolExecutor.java:895)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:918)
	at java.lang.Thread.run(Thread.java:662)
	at com.hazelcast.util.executor.HazelcastManagedThread.executeRun(HazelcastManagedThread.java:64)
	at com.hazelcast.util.executor.HazelcastManagedThread.run(HazelcastManagedThread.java:80)
```

--- a/hazelcast-client/src/test/java/com/hazelcast/client/cache/ClientCachingProviderTest.java
+++ b/hazelcast-client/src/test/java/com/hazelcast/client/cache/ClientCachingProviderTest.java
@@ -34,9 +34,12 @@ import org.junit.runner.RunWith;
 import javax.cache.spi.CachingProvider;
 import java.io.IOException;
 import java.util.ArrayList;
+import java.util.Collection;
 import java.util.Iterator;
 import java.util.List;
 
+import static com.hazelcast.cache.jsr.JsrTestUtil.clearCachingProviderRegistry;
+import static com.hazelcast.cache.jsr.JsrTestUtil.clearSystemProperties;
 import static org.junit.Assert.assertNotNull;
 
 @RunWith(HazelcastSerialClassRunner.class)
@@ -62,8 +65,6 @@ public class ClientCachingProviderTest extends CachingProviderTest {
             throw new AssertionError("Could not construct named hazelcast client instance: " +
                     e.getMessage());
         }
-        instances.add(instance1);
-        instances.add(instance2);
         instances.add(instance3);
         cachingProvider = createCachingProvider(instance1);
     }
@@ -89,6 +90,17 @@ public class ClientCachingProviderTest extends CachingProviderTest {
         HazelcastInstance otherInstance = HazelcastClient.getHazelcastClientByName(instanceName);
         assertNotNull(otherInstance);
         otherInstance.getLifecycleService().terminate();
+    }
+
+    @Override
+    protected Collection<HazelcastInstance> getStartedInstances() {
+        return HazelcastClient.getAllHazelcastClients();
+    }
+
+    @Override
+    protected void cleanupForDefaultCacheManagerTest() {
+        clearSystemProperties();
+        clearCachingProviderRegistry();
     }
 
     @After

--- a/hazelcast/src/test/java/com/hazelcast/cache/CachingProviderTest.java
+++ b/hazelcast/src/test/java/com/hazelcast/cache/CachingProviderTest.java
@@ -18,7 +18,6 @@ package com.hazelcast.cache;
 
 import com.hazelcast.cache.impl.AbstractHazelcastCachingProvider;
 import com.hazelcast.cache.impl.HazelcastServerCachingProvider;
-import com.hazelcast.cache.jsr.JsrTestUtil;
 import com.hazelcast.config.ClasspathXmlConfig;
 import com.hazelcast.config.Config;
 import com.hazelcast.core.Hazelcast;
@@ -39,11 +38,12 @@ import javax.cache.Caching;
 import javax.cache.spi.CachingProvider;
 import java.net.URI;
 import java.net.URISyntaxException;
-import java.util.Set;
+import java.util.Collection;
 
 import static com.hazelcast.cache.HazelcastCachingProvider.propertiesByInstanceItself;
 import static com.hazelcast.cache.HazelcastCachingProvider.propertiesByInstanceName;
 import static com.hazelcast.cache.HazelcastCachingProvider.propertiesByLocation;
+import static com.hazelcast.cache.jsr.JsrTestUtil.cleanup;
 import static org.junit.Assert.assertEquals;
 import static org.junit.Assert.assertNotNull;
 import static org.junit.Assert.assertTrue;
@@ -203,40 +203,40 @@ public class CachingProviderTest extends HazelcastTestSupport {
 
     @Test
     public void whenDefaultCacheManager_withUnnamedDefaultInstance_thenNoSharedNameHazelcastInstanceExists() {
-        JsrTestUtil.cleanup();
+        cleanupForDefaultCacheManagerTest();
         try {
             System.setProperty(AbstractHazelcastCachingProvider.NAMED_JCACHE_HZ_INSTANCE, "false");
             CachingProvider defaultCachingProvider = Caching.getCachingProvider();
             CacheManager defaultCacheManager = defaultCachingProvider.getCacheManager();
-            Set<HazelcastInstance> instances = Hazelcast.getAllHazelcastInstances();
+            Collection<HazelcastInstance> instances = getStartedInstances();
             for (HazelcastInstance instance : instances) {
-                if (AbstractHazelcastCachingProvider.SHARED_JCACHE_INSTANCE_NAME.equals(instance.getConfig().getInstanceName())) {
+                if (AbstractHazelcastCachingProvider.SHARED_JCACHE_INSTANCE_NAME.equals(instance.getName())) {
                     fail("The default named HazelcastInstance shouldn't have been started");
                 }
             }
             defaultCachingProvider.close();
         } finally {
-            JsrTestUtil.cleanup();
+            cleanup();
         }
     }
 
     @Test
     public void whenDefaultCacheManager_thenSharedNameHazelcastInstanceExists() {
-        JsrTestUtil.cleanup();
+        cleanupForDefaultCacheManagerTest();
         try {
             CachingProvider defaultCachingProvider = Caching.getCachingProvider();
             CacheManager defaultCacheManager = defaultCachingProvider.getCacheManager();
-            Set<HazelcastInstance> instances = Hazelcast.getAllHazelcastInstances();
+            Collection<HazelcastInstance> instances = getStartedInstances();
             boolean sharedInstanceStarted = false;
             for (HazelcastInstance instance : instances) {
-                if (instance.getConfig().getInstanceName().equals(AbstractHazelcastCachingProvider.SHARED_JCACHE_INSTANCE_NAME)) {
+                if (instance.getName().equals(AbstractHazelcastCachingProvider.SHARED_JCACHE_INSTANCE_NAME)) {
                     sharedInstanceStarted = true;
                 }
             }
             assertTrue("The default named HazelcastInstance should have been started", sharedInstanceStarted);
             defaultCachingProvider.close();
         } finally {
-            JsrTestUtil.cleanup();
+            cleanup();
         }
     }
 
@@ -248,5 +248,14 @@ public class CachingProviderTest extends HazelcastTestSupport {
         HazelcastInstance otherInstance = Hazelcast.getHazelcastInstanceByName(instanceName);
         assertNotNull(otherInstance);
         otherInstance.getLifecycleService().terminate();
+    }
+
+    protected Collection<HazelcastInstance> getStartedInstances() {
+        return Hazelcast.getAllHazelcastInstances();
+    }
+
+    // tests for the default cache manager require cleanup before running which must be overridden for client-side tests
+    protected void cleanupForDefaultCacheManagerTest() {
+        cleanup();
     }
 }

